django/db/models/sql/query.py

```diff
@@ -1296,8 +1296,8 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
                 #   <=>
                 # NOT (col IS NOT NULL AND col = someval).
                 lookup_class = targets[0].get_lookup('isnull')
-                col = _get_col(targets[0], join_info.targets[0], alias, simple_col)
-                clause.add(lookup_class(col, False), AND)
+                if condition.lhs.nullable:
```

Comment:
We can move check to the [previous `if` condition](https://github.com/django/django/pull/11600/files#diff-0edd853580d56db07e4020728d59e193R1286-R1288).